keiko-redis/src/main/kotlin/com/netflix/spinnaker/q/redis/RedisQueue.kt

```diff
@@ -402,6 +405,30 @@ class RedisQueue(
   )
 }
 
+private const val READ_MESSAGE = """
+  local java_scientific = function(x)
+    return string.format("%.12E", x):gsub("\+", "")
+  end
+
+  -- get the message, move the fingerprint to the unacked queue and return
+  local message = redis.call("HGET", messagesKey, fingerprint)
+
+  -- check for an ack timeout override on the message
+  local unackScore = unackDefaultScore
+  if message ~= nil then
+    local ackTimeoutOverride = tonumber(cjson.decode(message)["ackTimeoutMs"])
+    if ackTimeoutOverride ~= nil and unackBaseScore ~= nil then
```

Comment:
`tonumber` will return `nil` if its given any value it doesn't know how to process.